### ISSUE TITLE
Style transfer for a single font

### ISSUE THREAD

**Issue by YIYANGCAI** [clovaai/lffont#20]
Hi, thanks for this great work! I have implemented this code into a single font style transferring task. That means I only use one content font (6000+ chars) and a style font (500+ chars), and try to generate the rest chars of that style. When I trained the model for phase1, the output is really strange and the generator accuracy is really low (no more than 40%). Could you give some hints? I am looking forward to your reply and further discussions. 

Again, it is a fantastic work! Thank you very much!

Frank

**Comment by SanghyukChun** [comment]
Hi, thanks for the question.
Our method does not fit your scenario. Our method aims to train the model with various font styles, and generate a full font library with few-shot references (in our paper we show that only 8 references are enough).
Hence, my advice is just to apply our method to your style font. You don't have to choose a lot of references, but as in our paper, more references show better results. Thus, just download our model and try the few-shot generation without any training.

![image](https://user-images.githubusercontent.com/4919315/112179156-c7eca300-8c3d-11eb-8ab3-e6c3f66ec2ee.png)

However, our method is only applicable to the typeface fonts, but not applicable to colourful decorated images.
If you consider such images, our method will still give you disappointing results.

Another suggestion is to use [zi2zi](https://github.com/kaonashi-tyc/zi2zi), a very initial font generation method, but very effective if you have a lot of references as your case. It is trained with a lot of styles, and it **finetunes** the model with a lot of references with a coherent style, which is fit your case.

**Comment by YIYANGCAI** [comment]
> Hi, thanks for the question.
> Our method does not fit your scenario. Our method aims to train the model with various font styles, and generate a full font library with few-shot references (in our paper we show that only 8 references are enough).
> Hence, my advice is just to apply our method to your style font. You don't have to choose a lot of references, but as in our paper, more references show better results. Thus, just download our model and try the few-shot generation without any training. (https://github.com/clovaai/lffont/tree/master/weight)
> 
> ![image](https://user-images.githubusercontent.com/4919315/112179156-c7eca300-8c3d-11eb-8ab3-e6c3f66ec2ee.png)
> 
> However, our method is only applicable to the typeface fonts, but not applicable to colourful decorated images.
> If you consider such images, our method will still give you disappointing results.
> 
> Another suggestion is to use [zi2zi](https://github.com/kaonashi-tyc/zi2zi), a very initial font generation method, but very effective if you have a lot of references as your case. It is trained with a lot of styles, and it **finetunes** the model with a lot of references with a coherent style, which is fit your case.

Hi,

Thanks for your quick reply!

I just found out that the phase1 training finally generates a good result. In iteration before 37500, the validation results are really strange. However, the results after 40000 iteration become clear and acceptable. I suppose some modules in lffont is typically for multi-style but maybe it can still handle the single style problem? Before this I have also implemented the FUNIT to single style font generation and the result is similar.

**Comment by SanghyukChun** [comment]
@YIYANGCAI 
Our model is trained with 80,000 iterations. Maybe you need more iterations to get better results.
Especially, you need more iterations because our method has to learn component-wise features.
I suppose that your validation results are not good because the component-wise features are not well-trained yet.
Please try more iterations such as 20,000.

In theory, LF-Font can handle the single-style transfer as your task, but ours is specifically designed to handle multiple styles, that is why we need a matrix factorization module. Hence, if you use only a single style, maybe our method shows weak power than presented in our paper.

**Comment by YIYANGCAI** [comment]
Hi @SanghyukChun !

Thank you for the reply! I have implemented your pre-trained weights and the results are fairly acceptable. I am now trying something further. Could the weights be applied to a process of fine-tuning, and should I keep this gen's weights fixed for a number of iters and train the dis and cls first. and then training the entire model? 

Again, thanks for your reply and help to give me deeper understanding of this great work!

Frank.

**Comment by SanghyukChun** [comment]
@YIYANGCAI 
Disclaimer: we didn't test fine-tune approach with few-shot font generation frameworks before. Since our method is not designed for the fine-tune scenario, it may worse than you expected. IMHO, LF-Font could perform worse after the fine-tuning process.

In my previous experiences with [zi2zi](https://github.com/kaonashi-tyc/zi2zi), you don't have to freeze anything at the fine-tuning stage. But, as you know, we didn't upload any additional components such as discriminator or classifiers in this repository. I will check whether they are available to open in this repository.

**Comment by 8uos** [comment]
@YIYANGCAI 
Hi, you can get the weights of all modules here: (full model link)[https://drive.google.com/file/d/1zS5a49m_y5X2TUaRFWArLEgpQ7-RXq6N/view?usp=sharing].

**Comment by YIYANGCAI** [comment]
Hi!
Great! Thanks for the help!
